src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToLinqMethodProvider.cs

```diff
@@ -0,0 +1,169 @@
+                        case SyntaxKind.JoinClause: // Not supported because linq queries seem to provide essentially simpler syntax for the same than query methods.
+                        case SyntaxKind.FromClause: // More than one fromClause is not supported. The linq method seems to be more complicated for this.
+                        default: return null;
+                    }
+                }
+
+                var selectOrGroupClause = queryBody.SelectOrGroup;
+                switch (selectOrGroupClause.Kind())
+                {
+                    case SyntaxKind.SelectClause: expression = ProcesSelectClause(expression, context.Identifier, (SelectClauseSyntax)selectOrGroupClause); break;
+                    case SyntaxKind.GroupClause: expression = ProcessGroupClause(expression, context.Identifier, (GroupClauseSyntax)selectOrGroupClause); break;
+                    case SyntaxKind.LetClause: return null; // Skip this because the linq method for the let will be more complicated than the query.
+                    default: return null;
+                }
+
+                if (queryBody.Continuation != null)
+                {
+                    context = new ConversionContext(queryBody.Continuation.Identifier, expression);
+                    return ProcessQueryBody(context, queryBody.Continuation.Body);
+                }
+
+                return expression;
+            }
+
+            private static ExpressionSyntax ProcessGroupClause(ExpressionSyntax expression, SyntaxToken identifier, GroupClauseSyntax groupClause)
+            {
+                var parameter = SyntaxFactory.Parameter(identifier);
+                var groupExpressionLambda = SyntaxFactory.SimpleLambdaExpression(parameter, groupClause.GroupExpression.WithoutTrailingTrivia());
+                var byExpressionLambda = SyntaxFactory.SimpleLambdaExpression(parameter, groupClause.ByExpression.WithoutTrailingTrivia());
+                var groupExpressionArgument = SyntaxFactory.Argument(groupExpressionLambda);
+                var byExpressionArgument = SyntaxFactory.Argument(byExpressionLambda);
+                var arguments = new SeparatedSyntaxList<ArgumentSyntax>().Add(byExpressionArgument).Add(groupExpressionArgument);
+
+                return CreateInvocationExpression(expression, arguments, "GroupBy");
+            }
+
+            private static ExpressionSyntax ProcesSelectClause(ExpressionSyntax expression, SyntaxToken identifier, SelectClauseSyntax selectClause)
+            {
+                var selectExpression = selectClause.Expression;
+                // Avoid trivial Select(x => x)
+                if (selectExpression is IdentifierNameSyntax identifierName)
+                {
+                    // TODO consider a better condition to compare
+                    if (identifierName.Identifier.Text == identifier.Text)
```

Comment:
use ValueText instead of Text.  It normalizes names (i.e. erases unicode escapes).